src/core/ddsc/tests/filter.c

```diff
@@ -363,113 +363,6 @@ static bool filter_long1_eq_1 (const void *vsample)
   return sample->long_1 == 1;
 }
 
-CU_Test (ddsc_filter, compat)
-{
-  // check that the deprecated interface still works
-  dds_entity_t dp, tp, rd, wr;
-  dds_return_t ret;
-  char topicname[100];
-  create_unique_topic_name ("ddsc_filter", topicname, sizeof (topicname));
-  dds_qos_t *qos = dds_create_qos ();
-  dds_qset_reliability (qos, DDS_RELIABILITY_RELIABLE, DDS_INFINITY);
-  dds_qset_history (qos, DDS_HISTORY_KEEP_ALL, 0);
-  dp = dds_create_participant (0, NULL, NULL);
-  CU_ASSERT_FATAL (dp > 0);
-  tp = dds_create_topic (dp, &Space_Type1_desc, topicname, qos, NULL);
-  CU_ASSERT_FATAL (tp > 0);
-  DDSRT_WARNING_DEPRECATED_OFF;
-  dds_set_topic_filter (tp, filter_long1_eq_1);
-  DDSRT_WARNING_DEPRECATED_ON;
-  rd = dds_create_reader (dp, tp, qos, NULL);
-  CU_ASSERT_FATAL (rd > 0);
-  wr = dds_create_writer (dp, tp, qos, NULL);
-  CU_ASSERT_FATAL (wr > 0);
-  dds_delete_qos (qos);
-
-  ret = dds_write (wr, &(Space_Type1){0,0,0});
-  CU_ASSERT_FATAL (ret == 0);
-  ret = dds_write (wr, &(Space_Type1){1,0,0});
-  CU_ASSERT_FATAL (ret == 0);
-
-  struct exp exp = {
-    .n = 1, .xs = (const Space_Type1[]) {
-      {1,0,0}
-    },
-  };
-  checkdata (rd, &exp, "rd");
-  dds_delete (dp);
-}
-
-CU_Test (ddsc_filter, get)
```

Comment:
This one shouldn't go away in its entirety: `dds_get_topic_filter_extended` is still around and this seems to cover some cases not covered elsewhere. Perhaps you could double-check?

Reply:
I think that all that was tested in `ddsc_filter.get` with respect to `dds_get_topic_filter_extended` is covered in the test `ddsc_filter.getset_extended`, so that was the reason for removing the former. But I may be overlooking something?